pkg/internal/exec/file.go

```diff
@@ -75,7 +75,10 @@ func FindExecELF(p *services.ProcessInfo, svcID svc.ID) (*FileInfo, error) {
 
 	file.Service.EnvVars = envVars
 	if svcName, ok := file.Service.EnvVars[envServiceName]; ok {
-		file.Service.Name = svcName
+		// If Kubernetes is enabled we use the K8S metadata as the source of truth
+		if !k8sEnabled {
+			file.Service.Name = svcName
+		}
```

Comment:
We want to parse the rest of the variables, because the resource variables are needed for custom OTel tags customers provide.